feat: Document the URL escape for device command name and resource name (#980)
edgexfoundry/edgex-docs: docs_src/getting-started/Ch-GettingStartedDTOValidation.md

     - In BACNet protocol, the user might combine object type and property as the resourceName, for example, analog_input_0:present-value
     - In OPC_UA protocol, the user might use NodeId as the resourceName, for example, ns=10;s=Hello:World

!!! edgey "EdgeX 3.0"

>     In EdgeX 3.0 the character restriction was reduced for the command name and resource name because some protocol would use `/` or `.` for the resource name. By using the URL escape for API, device command name and resource name can allow lots of characters. for example, the user can define the command name `line-a/test:value` and use it with URL escape like `/api/v2/device/name/Modbus-TCP-Device/line-a%2Ftest%3Avalue`.

Reviewer: ```suggestion
    In EdgeX 3.0, the character restriction was reduced for the command name and resource name because some protocols may use `/` or `.` in the name. By using URL escaping for the API, device command name and resource name allow various characters. For example, the user can define the command name `line-a/test:value` and use it with URL escaping as `/api/v2/device/name/Modbus-TCP-Device/line-a%2Ftest%3Avalue`.
```

Author: Updated.
